src/pages/studyView/resources/FilesAndLinks.tsx

```diff
@@ -112,13 +125,29 @@ async function fetchFilesLinksData(
         0
     );
 
-    const resourcesForPatients = await getResourceDataOfPatients(
-        studyClinicalDataResponse.data
+    const selectedStudyIds = [
+        ...new Set(selectedSamples.map(item => item.studyId)),
+    ];
+
+    // sampleIds (+patientIds) for the selectedSamples
+    const selectedIds = new Map([
+        ...new Map(selectedSamples.map(item => [item.sampleId, item.studyId])),
+        ...new Map(selectedSamples.map(item => [item.patientId, item.studyId])),
+    ]);
+
+    // Fetch resources for entire study
+    const resourcesForEntireStudy = await getResourceDataOfEntireStudy(
+        selectedStudyIds
     );
-    const resourcesForPatientsAndSamples: { [key: string]: ResourceData[] } = {
-        ...sampleIdResourceData,
-        ...resourcesForPatients,
-    };
+
+    // Filter the resources to consist of only studyView selected samples
+    // Also keep patient level resources (e.g. Those don't have a sampleId)
+    const resourcesForPatientsAndSamples = _(resourcesForEntireStudy)
+        .filter(resource =>
+            selectedIds.has(resource.sampleId || resource.patientId)
+        )
+        .groupBy('patientId')
```

Comment:
@hweej one more little tweak.  i'm pretty sure that using the shortcut 'patientId' is not typesafe., so as a rule we use 
```
(r)=>r.patientId)
```
That way  if there is no propertyId, ts will catch it.